src/backend/columnar/columnar_tableam.c

```diff
@@ -2135,6 +2136,52 @@ ColumnarProcessUtility(PlannedStmt *pstmt,
 		RelationClose(rel);
 	}
 
+	if (IsA(parsetree, CreateExtensionStmt))
+	{
+		CreateExtensionStmt *createExtensionStmt = castNode(CreateExtensionStmt,
+															parsetree);
+		if (strcmp(createExtensionStmt->extname, "citus_columnar") == 0)
+		{
+			DefElem *newVersionValue = GetExtensionOption(createExtensionStmt->options,
+														  "new_version");
+			if (newVersionValue)
+			{
+				const char *newVersion = defGetString(newVersionValue);
+				if (strcmp(newVersion, "11.1-0") == 0)
+				{
+					ereport(ERROR, (errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+									errmsg("unsupported citus_columnar version 11.1-0")));
+				}
+			}
+
+			/*latest citus requires install columnar first, existing citus can only be an older version */
+			if (get_extension_oid("citus", true) != InvalidOid)
+			{
+				ereport(ERROR, (errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+								errmsg("must upgrade citus to version 11.1-1 first")));
+			}
```

Comment:
This is when cx installed a lower version of citus, and still want to install citus_columnar, if move to citus side, it could becomes:
1.  In citus utility_hook, check if it's install citus_columnar ? 
2.  Check the current installed citus version ?
3.  last throw error if citus version < 11.1

Reply:
done